src/index.d.ts

```diff
@@ -6,7 +6,14 @@ export function props(elem: any, data?: PropData): void;
 
 export const emit: (elem: any, eventName: string, eventOptions = {}) => boolean;
 
-export class Component extends HTMLElement {}
+export class Component extends HTMLElement {
+  updatedCallback(prevProps: { [name: string]: any }): boolean;
+  renderCallback(): any | undefined;
+  renderedCallback(): void;
+  connectedCallback(): void;
+  disconnectedCallback(): void;
```

Comment:
there absolutely no reason to do `any | any[]`, because it merges just to `any`.

Those definitions needs to be rewritten with appropriate `JSX` namespace but that's not part of this PR